src/osp/Resource/resourcetypes.h

```diff
@@ -93,19 +94,40 @@ struct res_container
 template <typename T>
 using res_container_t = typename res_container<T>::type;
 
+// Resource Owner
+
 class Resources;
 
-using ResIdStorage_t = lgrn::IdStorage<ResId, Resources>;
+using ResIdOwner_t = lgrn::IdStorage<ResId, Resources>;
+
+
+// Resource Type Ids
+
+struct GenResTypeId
+{
+    friend inline ResTypeId resource_type_next() noexcept;
+    friend inline int resource_type_count() noexcept;
+private:
+    static inline int count{0};
+};
+
+inline ResTypeId resource_type_next() noexcept
+{
+    return ResTypeId(GenResTypeId::count++);
```

Comment:
starting at zero is desired?

Means you can't detect if you've overflowed and rolled over.

Reply:
Yes, zero should be the first value returned. It's incredibly unlikely this is going to go any higher than 20 or so; overflow isn't a problem.